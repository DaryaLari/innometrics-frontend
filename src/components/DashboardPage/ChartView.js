import React from "react";
import styles from "./style.css";
import { Bar, BarChart, CartesianGrid, ResponsiveContainer, Tooltip, XAxis, YAxis } from 'recharts'

class ChartView extends React.Component {
  state = {
    activities: this.props.activities.map(a => {
      return {
        actName: a.executable_name,
        duration: Date.parse(a.start_time) - Date.parse(a.end_time)
      }})
  }
  render() {
    return (
      <div className={styles.chartView}>
        <ResponsiveContainer width={'100%'} aspect={2}>
          <BarChart data={this.state.activities}
                         margin={{ top: 25, right: 50, left: 50, bottom: 25 }}>
          <CartesianGrid strokeDasharray="3 3" />
          <XAxis dataKey="actName" label={{ offset: 0, value: "Activity", position: 'bottom' }} height={10}  axisLine={false} tickLine={false} tick={false} />
          <YAxis label={{ offset: 13, value: "Duration (sec)", position: 'top' }} dataKey="duration" />
          <Tooltip payload={this.state.activities} />
          <Bar dataKey="duration" barSize={30} fill="#8884d8" />
        </BarChart>
        </ResponsiveContainer>
      </div>
    )
  }
}

export default ChartView;