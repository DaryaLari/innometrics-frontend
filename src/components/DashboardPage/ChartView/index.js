import React from 'react'
import { connect } from 'react-redux'
import { Bar, BarChart, CartesianGrid, ResponsiveContainer, Tooltip, XAxis, YAxis } from 'recharts'
import { activitiesSummarized } from '../../../helpers/selectors'
import styles from './style.css'

class ChartView extends React.Component {
  render() {
    return (
      <div className={styles.chartView}>
        <ResponsiveContainer width={'100%'} aspect={2}>
          <BarChart data={this.props.activities}
                    margin={{ top: 25, right: 50, left: 50, bottom: 25 }}>
            <CartesianGrid strokeDasharray='3 3' />
            <XAxis dataKey='executable_name'
                   label={{ offset: 0, value: 'Activity', position: 'bottom' }}
                   height={25}
                   axisLine={false}
                   tickLine={false}
                   interval={0}
                   tick={<CustomTick amount={this.props.activities.length}/>}
            />
            <YAxis label={{ offset: 13, value: 'Duration (sec)', position: 'top' }} dataKey='duration' />
            <Tooltip payload={this.props.activities} />
            <Bar dataKey='duration'
                 barSize={30}
                 fill='#8884d8'
                 onClick={this.props.onBarClick}
            />
        </BarChart>
        </ResponsiveContainer>
      </div>
    )
  }
}

class CustomTick extends React.Component{
  render(){
    const {x, y, payload, ...rest} = this.props
    let width = rest.width/rest.amount - 5
    return (
      <foreignObject className={styles.truncated}
                     x={x-width/2} y={y-10}
                     width={width} height={20}
      >
        <span title={payload.value}>{payload.value}</span>
      </foreignObject>

    )
  }
}

const ConnectedChartView = connect(
  (state) => ({
    activities: activitiesSummarized(state)
  })
)(ChartView)

export default ConnectedChartView