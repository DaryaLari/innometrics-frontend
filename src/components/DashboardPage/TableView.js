import React from "react";
import styles from "./style.css";

const TableView = ({activities}) => {
  return (
    <div className={styles.tableView}>
      <table>
        <thead>
        <tr>
          <th>id</th>
          <th>user</th>
          <th>browser_title</th>
          <th>browser_url</th>
          <th>start_time</th>
          <th>end_time</th>
          <th>ip_address</th>
          <th>mac_address</th>
          <th>executable_name</th>
        </tr>
        </thead>
        <tbody>
        {activities.map(a => (
          <tr key={a._id}>
            <td>{a._id}</td>
            <td>{a.user}</td>
            <td>{a.browser_title}</td>
            <td>{a.browser_url}</td>
            <td>{a.start_time}</td>
            <td>{a.end_time}</td>
            <td>{a.ip_address}</td>
            <td>{a.mac_address}</td>
            <td>{a.executable_name}</td>
          </tr>
        ))}
        </tbody>
      </table>
    </div>
  )
}

export default TableView;