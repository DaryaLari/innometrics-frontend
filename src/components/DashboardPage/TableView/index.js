import React from "react";
import { connect } from 'react-redux'
import _ from 'lodash'
import { getActivities } from '../../../helpers/selectors'
import styles from "./style.css";

class TableView extends React.Component {
  state = {
    sortKey: null,
    order: null
  }
  sortBy = (key) => {
    if(this.state.order === null || this.state.order === 'desc'){
      this.setState({sortKey: key, order: 'asc'})
    }
    else{
      this.setState({sortKey: key, order: 'desc'})
    }
  }
  /*
  * @param { string } key - key of activities, by which the array should be sorted
  * @param { ['asc' | 'desc'] } order - for [ ascending or descending ] sorting order
  * */
  sortActivities = (key=this.state.key, order=this.state.order) => {
    if(key === null) {
      return this.props.activities
    }
    let activities = this.props.activities
    let modifiedKey = this.state.sortKey

    // if sortBy time, then convert value to data
    if(key === 'start_time' || key === 'end_time'){
      modifiedKey = (a) => Date.parse(a[key])
    }

    activities = _.sortBy(activities, modifiedKey)
    if(order === 'desc'){
      _.reverse(activities)
    }
    return activities
  }
  orderSign = (key) => {
    if(this.state.sortKey !== key){
      return ""
    }
    if(this.state.order === 'asc'){
      return <i className={`${"material-icons"} ${styles.sortDirection}`}>arrow_drop_up</i>
    }
    return <i className={`${"material-icons"} ${styles.sortDirection}`}>arrow_drop_down</i>
  }
  render(){
    let activities = this.sortActivities()
    return (
      <div className={styles.tableView}>
        <table className={styles.table}>
          <thead>
          <tr className={styles.row}>
            <th className={styles.cell}
                onClick={() => this.sortBy('start_time')}
            >
              Start time
              {this.orderSign('start_time')}
            </th>
            <th className={styles.cell}
                onClick={() => this.sortBy('end_time')}
            >
              End time
              {this.orderSign('end_time')}
            </th>
            <th className={styles.cell}
                onClick={() => this.sortBy('executable_name')}
            >
              File name
              {this.orderSign('executable_name')}
            </th>
          </tr>
          </thead>
          <tbody>
          {activities.map(a => (
            <tr key={a._id}
                className={`${styles.row} ${this.props.selectedActivity === a.executable_name && styles.selectedActivity}`}
            >
              <td className={styles.cell}>{a.start_time}</td>
              <td className={styles.cell}>{a.end_time}</td>
              <td className={styles.cell} title={a.executable_name}>{a.executable_name}</td>
            </tr>
          ))}
          </tbody>
        </table>
      </div>
    )
  }
}

const ConnectedTableView = connect(
  (state) => ({
    activities: getActivities(state)
  })
)(TableView)

export default ConnectedTableView;