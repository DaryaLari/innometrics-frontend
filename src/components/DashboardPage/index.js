import React from "react";
import {connect} from "react-redux";
import {getActivitiesRequest} from "../../store/activities/actionCreators";
import styles from "./style.css";

class DBPage extends React.Component {
  componentDidMount(){
    this.props.getActivities()
  }
  render() {
    return (
        <div className={styles.content}>
          <h1>Dashboard</h1>
        </div>

    )
  }
}

const DashboardPage = connect(
  (state) => ({
    activities: state.activities
  }),

  (dispatch) => ({
    getActivities: () => dispatch(getActivitiesRequest())
  })
)(DBPage)
