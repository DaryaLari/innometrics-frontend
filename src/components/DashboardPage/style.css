.content {

}

.commonView {
    display: flex;
    justify-content: stretch;
}

.truncated {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

table {
    table-layout: fixed;
}

th,td:nth-child(1){
    width: 50px;
}
th,td:nth-child(1){
    width: 10px;
}


.tableView {
    width: 50%;
    overflow: scroll;
}

.chartView {
    width: 50%;
}
