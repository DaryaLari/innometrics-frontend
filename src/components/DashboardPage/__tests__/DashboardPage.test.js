import React from 'react'
import createStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { shallow } from 'enzyme'
import DashboardPage from '../index'

describe('Test DashboardPage component', () => {

  it('Test shallow render DashboardPage component', () => {

    const mockStore = createStore([])
    const store = mockStore({})

    const wrapper = shallow(
      <Provider store={store}>
        <DashboardPage />
      </Provider>
    )

    expect(wrapper).toMatchSnapshot()
  })

})