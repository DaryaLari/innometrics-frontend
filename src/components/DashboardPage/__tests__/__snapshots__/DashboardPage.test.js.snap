// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DashboardPage component Test shallow render DashboardPage component 1`] = `<Connect(_DashboardPage) />`;
